influxdb/client.go

```diff
@@ -337,28 +336,26 @@ func (c *HTTPClient) Write(bp BatchPoints) error {
 	v.Set("consistency", bp.WriteConsistency())
 	u.RawQuery = v.Encode()
 
-	reqBody := io.Reader(&b)
-
 	if c.compression == "gzip" {
-		var err error
-		reqBody, err = CompressWithGzip(reqBody, gzip.DefaultCompression)
-		if err != nil {
-			return err
+		bodyWriteCloser := gzip.NewWriter(b)
+		for _, p := range bp.Points() {
+			if _, err := bodyWriteCloser.Write(p.BytesWithLineFeed(precision)); err != nil {
+				return err
+			}
 		}
-
-	}
-
-	for _, p := range bp.Points() {
-		if _, err := b.Write(p.Bytes(precision)); err != nil {
+		if err := bodyWriteCloser.Close(); err != nil {
 			return err
 		}
 
-		if err := b.WriteByte('\n'); err != nil {
```

Comment:
It's very surprising to me that this would be allocating - what is the allocation caused by?

Reply:
That line doesn't allocate, but the gzip writer doesn't have a WriteByte, it only has Write, which means allocating a single element byte slice.